backends/opt/ceed-opt-operator.c

```diff
@@ -193,18 +225,36 @@ static int CeedOperatorSetup_Opt(CeedOperator op) {
 
   // Set up infield and outfield pointer arrays
   // Infields
-  ierr = CeedOperatorSetupFields_Opt(qf, op, 0, blksize, impl->blkrestr,
-                                     impl->evecs, impl->evecsin,
-                                     impl->qvecsin, 0,
-                                     numinputfields, Q);
+  ierr = CeedOperatorSetupFields_Opt(qf, op, 0, 1, blksize,
+                                     impl->blkrestr, impl->evecs,
+                                     impl->evecsin, impl->qvecsin,
+                                     0, numinputfields, Q);
   CeedChk(ierr);
   // Outfields
-  ierr = CeedOperatorSetupFields_Opt(qf, op, 1, blksize, impl->blkrestr,
-                                     impl->evecs, impl->evecsout,
-                                     impl->qvecsout, numinputfields,
-                                     numoutputfields, Q);
+  ierr = CeedOperatorSetupFields_Opt(qf, op, 1, !identityQF, blksize,
+                                     impl->blkrestr, impl->evecs,
+                                     impl->evecsout, impl->qvecsout,
+                                     numinputfields, numoutputfields, Q);
   CeedChk(ierr);
 
+  // If identity q-function, input = output
+  if (identityQF) {
+    for (int i=0; i<numinputfields; i++) {
+      CeedEvalMode inmode;
+      ierr = CeedQFunctionFieldGetEvalMode(qfinputfields[i], &inmode);
+      CeedChk(ierr);
+      CeedEvalMode outmode;
+      ierr = CeedQFunctionFieldGetEvalMode(qfoutputfields[i], &outmode);
+      CeedChk(ierr);
+      if (inmode == CEED_EVAL_NONE && outmode == CEED_EVAL_NONE) {
+        return CeedError(ceed, 1,
+                         "CEED_EVAL_NONE for a matching input and output does not make sense with identity QFunction");
+      } else {
+        impl->qvecsout[i] = impl->qvecsin[i];
```

Comment:
I set the output qvecs and input qvecs to be the same to avoid copying

Reply:
Why wouldn't you do a `CEED_EVAL_NONE` with memcpy instead?

Reply:
I'm trying to avoid a memcopy altogether.

Reply:
What I meant is that I think that applying `Restriction`+`Prolongation` and doing nothing for the identity qFunction is more expensive than memcpy the input to the output (without `Restriction`+`Prolongation`). If it's really an identity operator, then the fastest should be memcpy, or am I missing something?

Reply:
There is still a basis action. The QFunction is identity, but the Operator is not. I have two operators in multigrid that do something with an identity QFunction.

Interpolate Operator:
Restrict Coarse, Basis Interp Coarse to Fine, QFunction Identity, Basis Eval None, Prolongate Fine
Restriction Operator:
Restrict Fine, Basis Interp Fine to Coarse, QFunction Identity, Basis Eval None, Prolongate Coarse

Currently CEED_EVAL_NONE is a no operation event in OpApply; I use the Qvec as the Evec, or vice versa. This PR extends that idea so I don't do a QFunction that is basically a memcopy for no reason when I can use the input Qvecs as the output Qvecs.